src/app/users/reset-password/reset-password.component.ts

```diff
@@ -0,0 +1,106 @@
+import {Component, ElementRef, OnInit, ViewChild} from '@angular/core';
+import {AccountNotFoundError, IncorrectPasswordFoundError, StudentService} from '../../services/Services';
+import {Router} from '@angular/router';
+import {CookieService} from '../../services/cookie/cookie.service';
+import { MessageService } from 'primeng';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './reset-password.component.html',
+  styleUrls: ['../../../animations.css', './reset-password.component.css']
+})
+export class ResetPasswordComponent implements OnInit {
+  CHANGE_NOTIFY_KEY = 'password-change-notify';
+  title = 'appName';
+
+  errorMessage = '';
+
+  constructor(private studentService: StudentService,
+              private router: Router,
+              private cookieService: CookieService,
+              private messageService: MessageService) {
+  }
+
+  @ViewChild('oldPassword') oldPasswordField: ElementRef;
+  @ViewChild('newPassword') newPasswordField: ElementRef;
+
+  ngOnInit(): void {
+    if (this.studentService.hasLogin()) {
+      this.authenticateWithCookie();
+    }
+  }
+
+  private authenticateWithCookie() {
+    const token = this.cookieService.get(StudentService.KEY_TOKEN);
+    if (token) {
+      this.studentService.auth(token).toPromise()
+        .then(s => { })
+        .catch(err => {
+          console.error(err);
+        });
+    }
+  }
```

Comment:
I think this method is duplicate to the one in the `loginComponent`

One common practice is to directly extract the common auth procedures to new classes. (in this case, `AuthenticationProcedure.ts`)
With the help of dependency-injection, you can easily reuse this procedure. As below

```js

export class AuthenticationProcedure  {
  constructor(private studentService: StudentService,
              private cookieService: CookieService) {
  }
 public authenticateWithCookieIfHasLogin() {
     if (this.studentService.hasLogin()) {
          this.authenticateWithCookie();
     }
 }

  public authenticateWithCookie() {
    const token = this.cookieService.get(StudentService.KEY_TOKEN);
    if (token) {
      this.studentService.auth(token).toPromise()
        .then(s => { })
        .catch(err => {
          console.error(err);
        });
    }
  }
```

put this file in a proper place

Remember to declare it in our `app.module` for dependency-injecting this class.

```js
export class ResetPasswordComponent implements OnInit {
...

  constructor(private studentService: StudentService,
              private router: Router,
              private authenticationProcedure: AuthenticationProcedure,
              private messageService: MessageService) {
  }
  ngOnInit(): void {
    authenticationProcedure.authenticateWithCookieIfHasLogin();
  }
...
```

Reply:
Angular is reporting `The class 'AuthenticationProcedure' is listed in the declarations of the NgModule 'AppModule', but is not a directive, a component, or a pipe. Either remove it from the NgModule's declarations, or add an appropriate Angular decorator.`
I think I will need some keyword to fix this issue.

Reply:
I think you shouldn't put it under the declarations

it is as similar as those `services` which are pure objects
and those services are put under the `providers` section

maybe add a element
`{provide: AuthenticationProcedure, useClass: AuthenticationProcedure}` can help